tests/metal/sv_target-complex-1.slang

```diff
@@ -0,0 +1,34 @@
+//TEST:SIMPLE(filecheck=CHECK): -target metal
+//TEST:SIMPLE(filecheck=CHECK-ASM): -target metallib
+
+struct NestedReturn
+{
+    float4 debug;
+};
+struct NestedReturn2
+{
+    float4 debugAux1;
+    float4 debugAux2;
+};
+
+// Semantics are supposed to ignore uppercase/lowercase differences
+struct Output
+{
+    float4 Diffuse : SV_TarGet0;
```

Comment:
The layout has overlap here, this should be an error instead of us silently adjusting things to make them non-overlapping.